Tests/xcodeprojMeasurementsTests/MeasurementsTests.swift

```diff
@@ -0,0 +1,97 @@
+import Foundation
+import PathKit
+import SwiftShell
+@testable import xcodeproj
+import XCTest
+
+final class MeasurementsTests: XCTestCase {
+    
+    let tempDirectory = Path.temporary
+    var projectPath: Path?
+    var sourceRoot: Path?
+    
+    override func setUp() {
+        super.setUp()
+        try? tempDirectory.delete()
+        self.projectPath = tempDirectory + Path("Project")
+        self.sourceRoot = tempDirectory + Path("Sources")
+        let projectPath = self.projectPath!
+        let sourceRoot = self.sourceRoot!
+        let xcodeproj = try! generateXcodeProj(path: projectPath, sourceRoot: sourceRoot)
+        validate(xcodeproj: xcodeproj)
+    }
+    
+    override func tearDown() {
+        super.tearDown()
+        try? tempDirectory.delete()
+    }
+    
+    func test_measureProjectRead() {
+        let duration = try! measureDuration {
+            _ = try XcodeProj(path: self.projectPath!)
+        }
+        XCTAssertLessThan(duration, 3)
```

Comment:
Values ​​with a large margin for independence from hardware. If something goes wrong, the values ​​will increase exponentially and it will be noticeable.

Reply:
Would you mind adding that as a comment?